libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/PushNotificationManager.swift

```diff
@@ -84,3 +89,100 @@ extension PushNotificationManager {
+        return result.notificationTypes
+    }
+    
+    private func storeNotification(types: [NotificationType], with userDefaults: UserDefaults) {
+        do {
+            let encodedData = try JSONEncoder().encode(types)
+            userDefaults.set(encodedData, forKey: UserDefaultsKeys.cachedNotificationTypes)
+        } catch {
+            print("Failed to encode notification types: \(error.localizedDescription)")
+        }
+    }
+    
+    private func getCachedNotificationTypes(with userDefaults: UserDefaults) -> [NotificationType]? {
+        guard let encodedData = userDefaults.data(forKey: UserDefaultsKeys.cachedNotificationTypes) else {
+            return nil
+        }
+        do {
+            return try JSONDecoder().decode([NotificationType].self, from: encodedData)
+        } catch {
+            print("Failed to decode notification types: \(error.localizedDescription)")
+            return nil
+        }
+    }
+    
+        private func setNotificationCategories(types: [NotificationType]) {
+            let categories = types.map { createNotificationCategory(from: $0) }
+            UNUserNotificationCenter.current().setNotificationCategories(Set(categories))
+        }
+    
+        private func createNotificationCategory(from type: NotificationType) -> UNNotificationCategory {
+            let actions = createActions(from: type.actionGroups)
+            return UNNotificationCategory(identifier: type.apiName, actions: actions, intentIdentifiers: [])
+        }
+    
+        private func createActions(from actionGroups: [ActionGroup]) -> [UNNotificationAction] {
+            return actionGroups.flatMap { actionGroup in
+                actionGroup.actions.map { action in
+                    UNNotificationAction(
+                        identifier: action.identifier,
+                        title: action.label,
+                        options: [.foreground] // Ensures the app opens if needed
+                    )
+                }
+            }
+        }
+}
+
+@objc
+extension PushNotificationManager {
+    /// Retrieves available notification types from the API or local cache if the API request fails.
+    ///
+    /// - Parameters:
+    ///   - restClient: The `RestClient` instance used to fetch notification types from the API.
+    ///   - userDefaults: Stores and retrieves cached notification types. Defaults to `.standard`.
+    /// - Returns: A tuple containing:
+    ///   - `Bool`: `true` if notification types were successfully retrieved and stored; `false` if retrieval failed.
+    ///   - `Error?`: An error object if retrieval fails completely, or `nil` if successful.
+    ///
+    /// If the API fetch fails, the function attempts to retrieve notification types from the cache.
+    /// If neither source provides data, the function returns `false` with an appropriate error.
+    @objc
+    public func getNotificationTypes(restClient: RestClient = RestClient.shared,
```

Comment:
This is the public method to call to get notification types.  Inside this method we store the types as well as set notification categores on UNUserNotificationCenter.